src/Contexts/AuthContext.js

```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useReducer } from "react";
+
+import { authReducer } from "../Reducers/AuthReducer";
+
+const AuthContext = createContext(null);
+
+const useAuth = () => useContext(AuthContext);
+
+const AuthContextProvider = ({ children }) => {
+  const oldToken = localStorage.getItem("token");
+
+  const [stateAuth, dispatchAuth] = useReducer(authReducer, {
+    userDetails: [],
+    token: oldToken ?? null,
+    isAuth: oldToken ? true : false,
```

Comment:
any specific reason to define isAuth ? u can validate whether user is signed up or not on the basis of token as well

Reply:
Just wanted to have a boolean instead of checking for token length